src/librustc_trans/partitioning.rs

```diff
@@ -312,7 +313,13 @@ fn place_root_translation_items<'a, 'tcx, I>(tcx: TyCtxt<'a, 'tcx, 'tcx>,
                     TransItem::Fn(ref instance) => {
                         let visibility = match instance.def {
                             InstanceDef::Item(def_id) => {
-                                if def_id.is_local() {
+                                let start_def_id = tcx.lang_items().require(StartFnLangItem);
+
+                                // If we encounter the lang start item, we set the visibility to
+                                // default.
+                                if start_def_id == Ok(def_id) {
```

Comment:
use `self.tcx().lang_items().start_fn == Some(def_id)` to avoid spurious errors on small tests.

Reply:
(To elaborate, I believe @arielb1's goal here is to avoid requiring the start lang item to be present if you are trying to write a tiny test targeting e.g. a `#![no_std]` library, where the start lang item would not otherwise be needed. I agree this makes sense.)

Reply:
Yeah, already done that! :)